src/frontends/lean/builtin_cmds.cpp

```diff
@@ -350,6 +350,14 @@ struct redirect_helper {
     }
 };
 
+static void read_pipe(lean::message_builder & out, object_ref & pipe) {
```

Comment:
The name is misleading. `pipe` here is a filehandle, and the function is copying the contents to `out`.